bin/resty

```diff
@@ -130,7 +130,15 @@ my $conns = $conns_num || 64;
 my @nameservers;
 
 if (!$resolve_ipv6) {
-    unshift @nameservers, "ipv6=off";
+    # no error check to let the later exec() fail.
+    # worst case: ipv6=off is not added to the nginx
+    # conf, but the worker won't spawn anyways...
+    my $stderr = `$nginx_path -V 2>&1 1>/dev/null`;
+    #warn $stderr;
+
+    if (index($stderr, "--with-ipv6") != -1) {
```

Comment:
According to Maxim Dounin's comments, we can - almost - safely assume that Nginx 1.11.5+ will be built with IPv6 support, since most modern OS will have support for it (I am only paraphrasing his observations here).

Hence, the `ipv6=off` option can safely be added in that case. This would give us the following behavior, updated from the previous comment:

* `< 1.11.5`
  * we perform the `--with-ipv6` check
    * support enabled
      * no `--resolve-ipv6` flag: we add the `ipv6=off` option
      * `--resolve-ipv6` flag: do nothing, it is nginx's default behavior
    * support disabled
      * no `--resolve-ipv6` flag: do nothing
      * `--resolve-ipv6` flag: error out: "No support for IPv6 resolution but --resolve-ipv6 flag"
* `>= 1.11.5`: assume IPv6 support enabled
  * no `--resolve-ipv6` flag: we add the `ipv6=off` option
  * `--resolve-ipv6` flag: do nothing, that is the default behavior when IPv6 support is enabled

This, of course, assumes that Maxim's comment is correct and that **all of OpenResty's supported platforms have support for IPv6**.

This would not require the current pending patch to the Nginx core at openresty/openresty#214 because we simply assume support for IPv6 and thus that the `ipv6=off` token will not throw an error.

Would that assumption be correct and should we be making it for future versions of OpenResty with the 1.11.5+ core?